iocore/dns/SplitDNS.cc

```diff
@@ -222,38 +222,36 @@ SplitDNS::findServer(RequestData *rdata, SplitDNSResult *result)
   /* ---------------------------
      the 'alleged' fast path ...
      --------------------------- */
-  if (m_bEnableFastPath) {
+  if (m_bEnableFastPath && m_pxLeafArray) {
     SplitDNSRecord *data_ptr = nullptr;
     char *pHost              = const_cast<char *>(rdata->get_host());
     if (nullptr == pHost) {
       Warning("SplitDNS: No host to match !");
       return;
     }
 
-    int len        = strlen(pHost);
-    HostLeaf *pxHL = static_cast<HostLeaf *>(m_pxLeafArray);
```

Comment:
I'm not familiar with `iocore/HostLookup.h`, but I'm pretty sure this cast is wrong.